cilium/CFP-28986-mutual-auth-updates.md

```diff
@@ -0,0 +1,256 @@
+* As soon as the destination node regains apiserver access, any ongoing, incorrectly allowed packet flow gets dropped even for established connections.
+
+
+However, if _all_ of those steps happen, then the cache manipulation involved will render it possible to spoof the mutual authentication process and allow unauthenticated pods to send traffic that the system will treat as though it's authenticated.
+
+Let's walk through some diagrams of this attack.
+
+![Node state before cache manipulation exploit](images/28986-before-exploit.png)
+
+Here we have two nodes of a Cilium cluster, meeting the above requirements.
+
+The ipcaches are synced between the two nodes, so all the identities are correct.
+
+Both nodes `A` and `B` agree on the IP addresses associated with the identites `foo` and `bar`.
+
+Once the exploit has been performed, the nodes look more like this:
+
+![Node state after cache manipulation exploit](images/28986-after-exploit.png)
+
+On Node `B`, the connection to the apiserver has been interrupted in some way, so the Cilium agent is no longer receiving updates to Cilium Endpoints, meaning that the ipcache will become out of sync.
+
+On node `A`, enough pods have been scheduled in a new identity (the `baz` identity), to have the IP address that was previously associated with `foo` be now associated with `baz`.
+
+However, on Node `B`, because the ipcache is out of sync, the ingress policy will allow traffic from the pod that's actually a `baz` pod, because it has incorrect information about the IP to identity mapping. This is the reason why the exploit only works in direct routing mode - because in encapsulated modes, the identity is passed in any packet and used directly, rather than relying on the ipcache.
+
+This CFP lays out a proposal to mitigate this risk in two ways:
+* Require the use of either encryption (recommended because it prevents man-in-the-middle attacks) or encapsulation (which prevents cache manipulation but not MITM). In both of these cases, the Cilium identity is included in every packet, and so the ipcache is not used to impute identity to packets.
+* Add an additional mutual authentication mode that will use the connection tracking ("conntrack") table in the conntrack BPF map to store information about what _connections_ need authentication, instead of what _identity-identity-node pairings_ need authentication.
+
+The first idea makes cache manipulation irrelevant, because the inbound packets include the security identity in a header.
+
+However, in the existing mode, mutual authentication handshakes are only initiated when there is a change in either endpoint state (that is, a new node to node connection from a new pod), or when a certificate is rotated. So, in the event that an attacker _did_ manage to compromise mutual authentication mechanisms somehow, the window is, at maximum, the certificate rotation period, which defaults to 30 minutes.
```

Comment:
I think we can talk about this earlier, in the description of the attack

Reply:
Added as per your other suggestion.